S2VX.Game/Leaderboard/AddLeaderboardEntryContainer.cs

```diff
@@ -0,0 +1,50 @@
+﻿using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Graphics.UserInterface;
+using osuTK;
+using osuTK.Graphics;
+
+namespace S2VX.Game.Leaderboard {
+    public class AddLeaderboardEntryContainer : FillFlowContainer {
+
+        private LeaderboardContainer LeaderboardContainer { get; }
+        private double Score { get; }
+        private BasicTextBox NameInput { get; set; }
+        private IconButton SaveButton { get; set; }
+
+        private const float InputWidth = 450.0f;
+        private const float InputHeight = 80.0f;
+
+        public AddLeaderboardEntryContainer(LeaderboardContainer leaderboardContainer, double score) {
+            LeaderboardContainer = leaderboardContainer;
+            Score = score;
+        }
+
+        [BackgroundDependencyLoader]
+        private void Load() {
+            var textSize = SizeConsts.TextSize1;
+            Width = InputWidth;
+            Height = InputHeight;
+            AutoSizeAxes = Axes.Both;
+            Children = new Drawable[] {
+                NameInput = new() {
+                    Size = new Vector2(InputWidth * 0.75f, InputHeight),
+                    BorderColour = Color4.Red,
+                    Masking = true,
+                },
+                SaveButton = new() {
+                    Action = () => {
+                        LeaderboardContainer.AddEntry(NameInput.Text, Score);
+                        Clear();
```

Comment:
This feels kinda awkward to me. Since disabling seems somewhat difficult, how about something like this:
- Always enable save button
- Remember entry once you save
- Remove previous entry if you already saved before

Also helps with cases where someone mistyped their name and wanted to correct it.

Reply:
Yeah this sounds like a good idea to me. The save button disappearing seemed ok to me but fixing typos would be nice.

Reply:
Implemented